server/protocols/jmap/src/test/java/org/apache/james/jmap/send/MailFactoryTest.java

```diff
@@ -1,123 +0,0 @@
-
-    private MailFactory testee;
-    private MetaDataWithContent message;
-    private Message jmapMessage;
-    private Envelope envelope;
-
-    @Before
-    public void init() throws MailboxException {
-        testee = new MailFactory();
-        String headers = "From: me@example.com\n"
-                + "To: 1@example.com\n"
-                + "Cc: 2@example.com, 3@example.com\n"
-                + "Bcc: 4@example.com\n"
-                + "Subject: news\n";
-        String content = headers
-                + "Hello! How are you?";
-        
-        message = MetaDataWithContent.builder()
-                .uid(MessageUid.of(2))
-                .keywords(Keywords.strictFactory().from(Keyword.SEEN))
-                .size(content.length())
-                .internalDate(Instant.now())
-                .sharedContent(new SharedByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8)))
-                .attachments(ImmutableList.of())
-                .mailboxId(InMemoryId.of(3))
-                .messageId(TestMessageId.of(2))
-                .build();
-        MessagePreviewGenerator messagePreview = mock(MessagePreviewGenerator.class);
-        HtmlTextExtractor htmlTextExtractor = mock(HtmlTextExtractor.class);
-        when(messagePreview.compute(any())).thenReturn("text preview");
-
-        MessageContentExtractor messageContentExtractor = new MessageContentExtractor();
-        BlobManager blobManager = mock(BlobManager.class);
-        when(blobManager.toBlobId(any(MessageId.class))).thenReturn(BlobId.fromString("fake"));
-        MessageFactory messageFactory = new MessageFactory(blobManager, messagePreview, messageContentExtractor, htmlTextExtractor);
-        jmapMessage = messageFactory.fromMetaDataWithContent(message);
-        envelope = Envelope.fromMessage(jmapMessage);
-    }
-
-    @Test(expected = NullPointerException.class)
-    public void buildMailShouldThrowWhenNullMailboxMessage() throws Exception {
-        testee.build(null, envelope);
-    }
-
-    @Test(expected = NullPointerException.class)
-    public void buildMailShouldThrowWhenNullJmapMessage() throws Exception {
-        testee.build(message, null);
-    }
-
-    @Test
-    public void buildMailShouldWork() throws Exception {
-        String expectedName = jmapMessage.getId().serialize();
-        MailAddress expectedSender = new MailAddress("me@example.com");
-        Collection<MailAddress> expectedRecipients = ImmutableSet.of(
-                new MailAddress("1@example.com"),
-                new MailAddress("3@example.com"),
-                new MailAddress("2@example.com"),
-                new MailAddress("4@example.com"));
-        
-        Mail actual = testee.build(message, envelope);
-        
-        assertThat(actual.getName()).isEqualTo(expectedName);
-        assertThat(actual.getMaybeSender()).isEqualTo(MaybeSender.of(expectedSender));
-        assertThat(actual.getRecipients()).containsAll(expectedRecipients);
-    }
```

Comment:
you do not keep these checks?

Reply:
We use MailImpl.Factory instead, which is fairly well tested

Reply:
there are some specific checks, such as a the message name is based on the message id. Do you think it's useless?

Reply:
I overlooked these tests, I put them back